parquet/src/main/java/org/apache/iceberg/data/parquet/BaseParquetWriter.java

```diff
@@ -50,6 +42,27 @@ protected ParquetValueWriter<T> createWriter(MessageType type) {
   protected abstract ParquetValueWriters.StructWriter<T> createStructWriter(
       List<ParquetValueWriter<?>> writers);
 
+  protected ParquetValueWriters.PrimitiveWriter<?> fixedWriter(ColumnDescriptor desc) {
+    return ParquetValueWriters.fixed(desc);
+  }
+
+  protected Optional<ParquetValueWriters.PrimitiveWriter<?>> dateWriter(ColumnDescriptor desc) {
+    return Optional.empty();
+  }
+
+  protected Optional<ParquetValueWriters.PrimitiveWriter<?>> timeWriter(ColumnDescriptor desc) {
+    return Optional.empty();
+  }
+
+  protected Optional<ParquetValueWriters.PrimitiveWriter<?>> timestampWriter(
+      ColumnDescriptor desc, boolean isAdjustedToUTC) {
+    return Optional.empty();
+  }
+
+  protected Optional<ParquetValueWriters.PrimitiveWriter<?>> uuidWriter(ColumnDescriptor desc) {
```

Comment:
Feedback here is similar to the read path:
* Should not use Optional
* Should produce the same writer as before
* Should deprecate this class so that we can make it package-private
* Should not need a factory method for UUID

Reply:
CI just failed for ArrowReaderTest > testReadAll().

Generic writer expects byte[] based UUID handling and internal reader and writer works directly on UUID object. So, we do need a factory method here as we need to supply different kinds of UUID representation.